messaging/src/main/java/org/axonframework/eventhandling/tokenstore/jpa/JpaTokenStore.java

```diff
@@ -227,6 +237,31 @@ protected TokenEntry loadToken(String processorName, int segment, EntityManager
         return token;
     }
 
+    @Override
+    public Optional<String> retrieveStorageIdentifier() {
+        try {
+            return Optional.of(getConfig()).map(i -> i.get("id"));
+        } catch (Exception e) {
+            throw new UnableToRetrieveIdentifierException("Exception occurred while trying to establish storage identifier", e);
+        }
+    }
+
+    private ConfigToken getConfig() {
+        EntityManager em = entityManagerProvider.getEntityManager();
+        TokenEntry token = em
+                .find(TokenEntry.class, new TokenEntry.PK(CONFIG_TOKEN_ID, CONFIG_SEGMENT), LockModeType.NONE);
+        if (token == null) {
+            token = new TokenEntry(CONFIG_TOKEN_ID, CONFIG_SEGMENT, new ConfigToken(Collections.singletonMap("id", UUID.randomUUID().toString())), serializer);
+            em.persist(token);
+            em.flush();
+        }
+        return (ConfigToken) token.getToken(serializer);
+    }
+
+    public Serializer serializer() {
```

Comment:
What do we need this method for? If it's here to stay, I think it deserves some Javadoc though.